packages/react/src/__tests__/ReactElementClone-test.js

```diff
@@ -274,8 +274,22 @@ describe('ReactElementClone', () => {
 
     const root = ReactDOMClient.createRoot(document.createElement('div'));
     await act(() => root.render(<Grandparent />));
-    expect(component.childRef).toEqual({current: null});
-    expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    if (gate(flags => flags.enableRefAsProp && flags.disableStringRefs)) {
+      expect(component.childRef).toEqual({current: null});
+      expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    } else if (
+      gate(flags => !flags.enableRefAsProp && !flags.disableStringRefs)
+    ) {
+      expect(component.childRef).toEqual({current: null});
+      expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
+    } else if (
+      gate(flags => flags.enableRefAsProp && !flags.disableStringRefs)
+    ) {
+      expect(component.childRef).toEqual({current: null});
+      expect(component.parentRef.current.xyzRef.current.tagName).toBe('SPAN');
```

Comment:
i'm pretty sure this isn't running...

Reply:
ah, not by default. thanks CI

Reply:
Nah, it's getting hit, you have to run with the www variant:

```
yarn test-www --variant=true
```

Tested locally:

<img width="835" alt="Screenshot 2024-04-09 at 3 01 39 PM" src="https://github.com/facebook/react/assets/2440089/2486a1e0-c24e-42b9-b457-7ddd0633c8d5">